applications/nrf5340_audio/src/bluetooth/le_audio_cis_gateway.c

```diff
@@ -265,23 +311,23 @@ static void work_stream_start(struct k_work *work)
 		LOG_ERR("Cannot get info for start stream");
 	}
 
-	ret = bt_audio_stream_start(&audio_streams[work_data.channel]);
+	ret = bt_audio_stream_start(&audio_streams[work_data.stream_index]);
 	work_data.retries++;
 
 	if (ret) {
 		if (work_data.retries < CIS_CONN_RETRY_TIMES) {
 			LOG_WRN("Got connect error from stream %d Retrying. code: %d count: %d",
-				work_data.channel, ret, work_data.retries);
+				work_data.stream_index, ret, work_data.retries);
 			ret = k_msgq_put(&kwork_msgq, &work_data, K_NO_WAIT);
 			if (ret) {
 				LOG_ERR("No space in the queue for the bond");
 			}
 			/* Delay added to prevent controller overloading */
-			k_work_reschedule(&stream_start_work[work_data.channel], K_MSEC(500));
+			k_work_reschedule(&stream_start_work[work_data.stream_index], K_MSEC(500));
 		} else {
-			LOG_ERR("Could not connect ch %d after %d retries", work_data.channel,
+			LOG_ERR("Could not connect ch %d after %d retries", work_data.stream_index,
 				work_data.retries);
-			bt_conn_disconnect(headset_conn[work_data.channel],
+			bt_conn_disconnect(audio_streams[work_data.stream_index].conn,
```

Comment:
Better to use headsets array in stead of audio_streams array here?

Reply:
See answer above